modules/json/format-json.c

```diff
@@ -333,15 +333,18 @@ tf_json_call(LogTemplateFunction *self, gpointer s,
              const LogTemplateInvokeArgs *args, GString *result)
 {
   TFJsonState *state = (TFJsonState *)s;
-  gint i;
-  gboolean r = TRUE;
   gsize orig_size = result->len;
 
-  for (i = 0; i < args->num_messages; i++)
-    r &= tf_json_append(result, state->vp, args->messages[i], args->opts, args->seq_num, args->tz);
+  for (gint i = 0; i < args->num_messages; i++)
+    {
+      gboolean r = tf_json_append(result, state->vp, args->messages[i], args->opts, args->seq_num, args->tz);
```

Comment:
it is possible to factor out this into a separate func:
```
static gboolean
_json_append(result, state, args)
{
...
}
```

```
for (gint i = 0; i < args->....; i++ )
  {
    if (!_json_append(result, state, args))
      return;
  }
```

Reply:
I would leave it as it is,  as the 2 _call functions use 2 different _append functions.
Yes, that's the only difference, but otherwise it would be more complex:
```
static gboolean
_json_append(TFJsonState *state, const LogTemplateInvokeArgs args, LogMessage *msg, 
             JsonAppendCb callback, GString *result)
{
....
}


  for (gint i = 0; i < args->num_messages; i++)
    {
      if(!_json_append(state, args, args->messages[i], tf_flat_json_append, result))
        return;
    }   
```